start.py

```diff
@@ -1,40 +1,100 @@
+            else:
+                response = False
+                while response is False:
+                    Ui.msg(BoardPresenter.display_terminal_board(current_state))
+                    response = CommandLinePrompt.get_input("Enter a number from 1-9: ")
+                    response = UserActions.make_move(current_state, response)
+                BoardState.update_state(current_state, User.current_player, response)
+
+                if EndStates.did_a_player_win(current_state, User.current_player, win_config.winning_combos):
+                    Ui.msg(BoardPresenter.display_terminal_board(current_state))
+                    Ui.msg('Game Over: ' + User.current_player + ' WINS!')
+                    return True
+
+                if EndStates.is_draw(current_state):
+                    Ui.msg('DRAW. GameOver')
+                    return True
+
+                if User.current_player == player1:
+                    User.switch_current_user(User.current_player, player2)
+                else:
+                    User.switch_current_user(User.current_player, player1)
+
+    def computer_v_computer(current_state, player1, player2):
+        game_over = False
+        while game_over is False:
+                Ui.msg(BoardPresenter.display_terminal_board(current_state))
+                computers_move = Ai.make_move(current_state)
+                BoardState.update_state(current_state, User.current_player, computers_move)
+                if EndStates.did_a_player_win(current_state, User.current_player, win_config.winning_combos):
+                    Ui.msg(BoardPresenter.display_terminal_board(current_state))
+                    Ui.msg('Game Over: ' + User.current_player + ' WINS!')
+                    return True
+
+                if EndStates.is_draw(current_state):
+                    Ui.msg('DRAW. GameOver')
+                    return True
+
+                if User.current_player == player1:
+                    User.switch_current_user(User.current_player, player2)
+                else:
+                    User.switch_current_user(User.current_player, player1)
+
+def begin_game(mode_choosen, current_state, player1, player2):
+    if mode_choosen == "1":
+       return GameModes.human_v_human(current_state, player1, player2)
+    if mode_choosen == "2":
+       return GameModes.human_v_computer(current_state, player1, player2)
+    if mode_choosen == "3":
+       return GameModes.computer_v_computer(current_state, player1, player2)
 
 if __name__ == "__main__":
     game_board = Board()
-    game_board.new_game(3) 
+    game_board.new_game(3)
     win_config = WinningCombos(3)
     win_config.create_winning_combos()
     player1 = User("X")
     player2 = User("O")
-    start_game(game_board.board_state, player1.symbol, player2.symbol)
+    mode_choice = UserActions.pick_game_mode()
+    begin_game(mode_choice, game_board.board_state, player1.symbol, player2.symbol)
```

Comment:
I suggest you start over. Start the game loop over with a clean slate and build it up with TDD. Ignore whatever other classes and files you have in place and free your self of any current preconceptions you have about how the game should be designed. You don't have to throw away your code, but I think it will be good to pretend like it's not there. If you can fully test your game loop the rest of the code will be created and tested out of necessity. Let me know if you have questions about this. I had to throw away a month of work on my HTTP server and start over in a new language it was an extremely valuable and rewarding experience.

Reply:
Completely agree